File: client_manage_interface/src/main/java/cn/edu/aynu/rjxy/management/service/CinemaService.java
package cn.edu.aynu.rjxy.management.service;

import cn.edu.aynu.rjxy.pojo.TbCinema;
import com.github.pagehelper.PageInfo;

import java.util.List;

/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface CinemaService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<TbCinema> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageInfo findPage(int pageNum, int pageSize);


	/**
	 * 增加
	*/
	public void add(TbCinema cinema);


	/**
	 * 修改
	 */
	public void update(TbCinema cinema);


	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public TbCinema findOne(Long cinemaId);


	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Long[] cinemaIds);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageInfo findPage(TbCinema cinema, int pageNum, int pageSize);

	void updateIsUsable(Long cinemaId, Long isUsable);
}

File: client_manage_web/src/main/webapp/js/service/cinemaService.js
//服务层
app.service('cinemaService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../cinema/findAll.do');		
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../cinema/findPage.do?pageNum='+page+'&pageSize='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../cinema/findOne.do?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../cinema/add.do',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../cinema/update.do',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../cinema/delete.do?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../cinema/search.do?pageNum='+page+"&pageSize="+rows, searchEntity);
	}
	this.updateIsUsable=function (cinemaId,isUsable) {
		return $http.get('../cinema/updateIsUsable.do?cinemaId='+cinemaId+'&isUsable='+isUsable);
	}
});